src/Tools/dotnet-monitor/Egress/EgressProviderSource.cs

```diff
@@ -98,6 +104,13 @@ private void Reload()
                     }
                 }
             }
+
+            CancellationTokenSource source = new(ValidationTimeoutMilliseconds);
+
+            foreach (var providerName in _providerNameToTypeMap.Keys)
```

Comment:
This isn't particularly safe to do from a non-async context and creates unobservable Tasks. Let's move this to a new implementation that derives from BackgroundService. Then add a Reloaded event to the EgressProviderSource that the background services can listen to. Each time the event is raised, the background service can then async await the validation of each provider. If validation is already in flight, then those need to be canceled before validating the reload.